sentry-core/src/client.rs

```diff
@@ -60,10 +62,15 @@ impl fmt::Debug for Client {
 impl Clone for Client {
     fn clone(&self) -> Client {
         let transport = Arc::new(RwLock::new(self.transport.read().unwrap().clone()));
+
+        #[cfg(feature = "release-health")]
         let session_flusher = RwLock::new(Some(SessionFlusher::new(
             transport.clone(),
             self.options.session_mode,
         )));
+        #[cfg(not(feature = "release-health"))]
+        let session_flusher = RwLock::new(None);
```

Comment:
it might be a bit more work, but avoiding the `Client.session_flusher` field altogether would be slightly more efficient.